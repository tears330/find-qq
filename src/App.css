.App {
  background-color: #282c34;
  height: 100vh;
  padding: 0 50px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  color: rgba(255, 255, 255, 0.8);
  height: 140px;
  font-size: 30px;
}

.App-link {
  color: #61dafb;
}

.result-section {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 50px;
  min-height: 200px;
}
