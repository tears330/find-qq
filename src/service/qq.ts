import { request } from '../utils';

export interface QqInfo {
  code: 1;
  qq: string;
  name: string;
  qlogo: string;
}

export interface ResponstError {
  code: 201702;
  msg: string;
}

export const fetchQqInfo = (qq: string) =>
  request
    .get(`/qq.info`, {
      params: { qq },
    })
    .then((res) => {
      const result = JSON.parse(res.data) as QqInfo | ResponstError;
      if (result.code === 1) {
        return result;
      } else {
        throw new Error(result.msg);
      }
    });
