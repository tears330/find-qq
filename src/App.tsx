import React, { ChangeEventHandler, FC, useCallback, useState } from 'react';
import { AccountCard } from './components/account-card';
import { Result } from './components/result';
import { SearchInput } from './components/search-input';
import { Spin } from './components/spin';
import { useDebounce, useRequest } from './hooks';
import { fetchQqInfo } from './service';

import './App.css';

export const App: FC = () => {
  const [searchValue, setSearchValue] = useState('');
  const debounceValue = useDebounce(searchValue, 800);

  const {
    data: accountInfo,
    loading,
    error,
  } = useRequest(fetchQqInfo, {
    params: debounceValue,
    paramsValidate: Boolean,
  });

  const handleSearch: ChangeEventHandler<HTMLInputElement> = useCallback(
    (e) => {
      setSearchValue(e.target.value.replace(/[^0-9]/g, ''));
    },
    [],
  );

  return (
    <div className="App">
      <header className="App-header">üïµÔ∏è QQ Âè∑Êü•ËØ¢</header>
      <section className="search-section">
        <SearchInput
          value={searchValue}
          onChange={handleSearch}
          type="search"
          maxLength={10}
        />
      </section>
      <Spin spinning={loading}>
        <section className="result-section">
          {accountInfo && accountInfo.code === 1 && (
            <AccountCard account={accountInfo} />
          )}
          {error && <Result msg={error.message || 'ÁΩëÁªúÂºÇÂ∏∏'} />}
        </section>
      </Spin>
    </div>
  );
};
