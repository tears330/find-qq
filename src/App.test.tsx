import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { App } from './App';
import { act } from 'react-dom/test-utils';
import { fetchQqInfo } from './service';

jest.mock('./service');

const mockFetchQqInfo = fetchQqInfo as typeof fetchQqInfo & jest.Mock;

describe('find qq', () => {
  it('basic render', async () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();

    mockFetchQqInfo.mockResolvedValue({
      name: 'test',
      qq: '123123',
      code: 1,
      qlogo: 'https://q2.qlogo.cn/headimg_dl?spec=100&dst_uin=123123',
    });

    const input = screen.getByRole('searchbox');

    fireEvent.input(input, { target: { value: '123123' } });

    act(() => {
      jest.advanceTimersByTime(800);
    });

    await waitFor(() => {
      expect(screen.getByText('test')).toBeInTheDocument();
    });

    expect(baseElement).toMatchSnapshot();

    mockFetchQqInfo.mockRejectedValue(new Error('error-test'));

    fireEvent.input(input, { target: { value: '123' } });

    act(() => {
      jest.advanceTimersByTime(800);
    });

    await waitFor(() => {
      expect(screen.getByText('error-test')).toBeInTheDocument();
    });

    expect(baseElement).toMatchSnapshot();

    mockFetchQqInfo.mockRejectedValue(new Error());

    fireEvent.input(input, { target: { value: '1233' } });

    act(() => {
      jest.advanceTimersByTime(800);
    });

    await waitFor(() => {
      expect(screen.getByText('网络异常')).toBeInTheDocument();
    });

    expect(baseElement).toMatchSnapshot();
  });
});
