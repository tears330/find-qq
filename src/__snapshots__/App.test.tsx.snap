// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find qq basic render 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        üïµÔ∏è QQ Âè∑Êü•ËØ¢
      </header>
      <section
        class="search-section"
      >
        <div
          class="com-search"
        >
          <input
            autocomplete="off"
            maxlength="10"
            type="search"
            value=""
          />
        </div>
      </section>
      <section
        class="result-section"
      />
    </div>
  </div>
</body>
`;

exports[`find qq basic render 2`] = `
<body>
  <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        üïµÔ∏è QQ Âè∑Êü•ËØ¢
      </header>
      <section
        class="search-section"
      >
        <div
          class="com-search"
        >
          <input
            autocomplete="off"
            maxlength="10"
            type="search"
            value="123123"
          />
        </div>
      </section>
      <section
        class="result-section"
      >
        <div
          class="com-account-card"
        >
          <img
            alt="123123"
            src="https://q2.qlogo.cn/headimg_dl?spec=100&dst_uin=123123"
          />
          <div
            class="com-account-card-info"
          >
            <div
              class="com-account-card-info-name"
            >
              test
            </div>
            <div
              class="com-account-card-info-qq"
            >
              123123
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`find qq basic render 3`] = `
<body>
  <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        üïµÔ∏è QQ Âè∑Êü•ËØ¢
      </header>
      <section
        class="search-section"
      >
        <div
          class="com-search"
        >
          <input
            autocomplete="off"
            maxlength="10"
            type="search"
            value="123"
          />
        </div>
      </section>
      <section
        class="result-section"
      >
        <div
          class="com-result"
        >
          <p>
            Woops! Something went wrong :(
          </p>
          <p>
            error-test
          </p>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`find qq basic render 4`] = `
<body>
  <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        üïµÔ∏è QQ Âè∑Êü•ËØ¢
      </header>
      <section
        class="search-section"
      >
        <div
          class="com-search"
        >
          <input
            autocomplete="off"
            maxlength="10"
            type="search"
            value="1233"
          />
        </div>
      </section>
      <section
        class="result-section"
      >
        <div
          class="com-result"
        >
          <p>
            Woops! Something went wrong :(
          </p>
          <p>
            ÁΩëÁªúÂºÇÂ∏∏
          </p>
        </div>
      </section>
    </div>
  </div>
</body>
`;
